app/controllers/application_controller.rb

```diff
@@ -38,12 +38,9 @@ def user_signed_in?
   helper_method :user_signed_in?
 
   def authenticate_user!
-    if current_user
-      true
-    else
-      session["return_to"] ||= request.url
-      redirect_to login_path unless login_redirect? or oauth_callback?
-    end
+    return true if current_user
+    session["return_to"] ||= request.url
+    redirect_to login_path unless login_redirect? or oauth_callback?
```

Comment:
it looks like the unless can just be taken off there, and just redirect_to login_path, but i'm not sure if that breaks anything